src/libexpr/pos-idx.hh

```diff
@@ -37,8 +40,28 @@ public:
     {
         return id == other.id;
     }
+
+    size_t hash() const noexcept
+    {
+        size_t h = 854125;
+        hash_combine(h, id);
```

Comment:
This `hash_combine()` seems unnecessary? We can just return the hash of `id`.

Reply:
In Java it's common practice to include a random number to represent the distinct type, but I think you're right.
That doesn't really apply here, because we don't have a "top" type like Java's `Object` that would create a need to avoid collisions when used in e.g. `HashMap<String, Object>`.
The closest thing we might have is `std::variant`. I think we can assume that it hashes its index number, which is probably better than salting each type anyway, because iirc it allows multiple distinct occurrences of the same type.